### PR TITLE
add assert for lb ub

### PR DESCRIPTION
**Pull request by AnasAbdelR**
Currently if lb and ub are switched, this will still silently fail. Adding an assert here will catch this and warn the user of the error.

### CODE REVIEW

**Comment by AnasAbdelR** [comment]
Closes #38

**Comment by ChrisRackauckas** [comment]
Asserts are for developers, not for users. This should probably be a proper error exception, like https://github.com/SciML/DiffEqBase.jl/blob/master/src/solve.jl#L116-L137

**Comment by AnasAbdelR** [comment]
Addressed in 15c170b

File: src/QuasiMonteCarlo.jl [hunk 1]
```diff
@@ -6,6 +6,16 @@ abstract type SamplingAlgorithm end
 
 include("Faure.jl")
 
+const UB_LB_MESSAGE = """
+                          "lb must be less than ub"
+                         """
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:11]: That seems formatted incorrectly. Did the formatter do that?

File: src/QuasiMonteCarlo.jl [hunk 2]
```diff
@@ -140,6 +150,9 @@ sample(n,lb,ub,S::GridSample)
 Returns a tuple containing numbers in a grid.
 """
 function sample(n, lb, ub, S::GridSample)
+     if !all(lb.<ub)
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:153]:
```suggestion
     if !all((x,y)->x<y,zip(lb,ub)) 
```

Non-allocating

Comment (ChrisRackauckas) [reply src/QuasiMonteCarlo.jl:153]: But pull that out to a separate small function so it precompiles.

Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl]: Addressed in 2ad2d19

File: src/QuasiMonteCarlo.jl (hunk 1 above)
Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl:12]: Addressed in f2fcc95

File: src/QuasiMonteCarlo.jl [hunk 3]
```diff
@@ -169,12 +186,15 @@ sample(n,lb,ub,::SobolSampling)
...
+        throw(UbLbWrong())
+    end
     s = SobolSeq(lb, ub)
     skip(s, n)
     if lb isa Number
-        return [next!(s)[1] for i in 1:n]
+        return [next!(s)[1] for i = 1:n]
     else
-        return reduce(hcat, [next!(s) for i in 1:n])
+        return reduce(hcat, [next!(s) for i = 1:n])
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:197]: why all of these `in` to `=` ? Is that a formatter thing?

File: src/QuasiMonteCarlo.jl [hunk 4]
```diff
@@ -412,11 +450,19 @@ function generate_design_matrices(n, lb, ub, sampler, num_mats = 2)
...
+    UniformSample,
+    SobolSample,
+    LatinHypercubeSample,
+    LatticeRuleSample,
+    RandomSample,
+    LowDiscrepancySample,
+    GoldenSample,
+    KroneckerSample,
+    SectionSample,
+    FaureSample
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:466]: Did you run the correct formatter?

**Comment by ChrisRackauckas** [comment]
This fails tests.

File: src/QuasiMonteCarlo.jl (hunk 4 above)
Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl]: Addressed in 28aaa9b

File: src/QuasiMonteCarlo.jl (hunk 3 above)
Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl]: Yes - wrong formatter - fixed. Addressed in 28aaa9b

File: src/QuasiMonteCarlo.jl (hunk 1 above)
Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl:12]: Fixed again, addressed in 28aaa9b

**Comment by AnasAbdelR** [comment]
Tests fail because lower and upper bound can be passed in as a scalar instead of a vector - breaking the check bounds. Should we fix the check bounds or change tests to be an array with a single element as opposed to a scalar?

**Comment by ChrisRackauckas** [comment]
Scalars are allowed. The check bounds code needs a dispatch for htat

**Comment by AnasAbdelR** [comment]
Addressed in c3a063c

File: src/QuasiMonteCarlo.jl [hunk 5]
```diff
@@ -415,8 +454,16 @@ function generate_design_matrices(n, lb, ub, sampler, num_mats = 2)
...
+export GridSample,
+       UniformSample,
+       SobolSample,
+       LatinHypercubeSample,
+       LatticeRuleSample,
+       RandomSample,
+       LowDiscrepancySample,
+       GoldenSample,
+       KroneckerSample,
+       SectionSample,
```
Comment (ChrisRackauckas) [inline src/QuasiMonteCarlo.jl:466]: why this? I don't think the formatter would do this.

File: test/runtests.jl [hunk 6]
```diff
@@ -1,5 +1,6 @@
 using QuasiMonteCarlo, Distributions, StatsBase
 using Test
+using QuasiMonteCarlo:check_bounds
```
Comment (ChrisRackauckas) [inline test/runtests.jl:3]:
```suggestion
using QuasiMonteCarlo: check_bounds
```

File: src/QuasiMonteCarlo.jl (hunk 5 above)
Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl:466]: I mean that's what it did 🤷🏼‍♂️

Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl:466]: Let me dig into it some more.

Comment (AnasAbdelR) [reply src/QuasiMonteCarlo.jl:466]: ` [98e50ef6] JuliaFormatter v1.0.2` is what I am using... not sure what is going on?

**Pull request merged**